File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        stack = []
        n = len(temperatures)
        res = [0] * n
        
        for i in range(0,n):
            while stack and temperatures[stack[-1]] < temperatures[i]:
                temp = stack.pop()
                warmer_day = i - temp
                res[temp] = warmer_day
            stack.append(i)
        return res

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        Stack<Integer> myStack = new Stack<>();
        int[] ans = new int[n];

        for(int i =0; i < n; i++){
            while(!myStack.isEmpty() && temperatures[myStack.peek()] < temperatures[i]){
                int previndex = myStack.pop();
                ans[previndex] = i - previndex;
            }
            myStack.push(i);
        }

        return ans;
    }
}